pkg/vpn_server/iptables.go

```diff
@@ -0,0 +1,55 @@
+package vpn_server
+
+import (
+	"fmt"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/go-logr/logr"
+
+	"github.com/gardener/vpn2/pkg/config"
+	"github.com/gardener/vpn2/pkg/constants"
+	"github.com/gardener/vpn2/pkg/network"
+)
+
+func SetIPTableRules(log logr.Logger, cfg config.VPNServer) error {
+	if !cfg.IsHA {
+		log.Info("setting up double NAT IPv4 iptables rules")
+		ipTable, err := network.NewIPTables(log, iptables.ProtocolIPv4)
+		if err != nil {
+			return err
+		}
+
+		ipv4PodNetworks := network.GetByIPFamily(cfg.PodNetworks, network.IPv4Family)
+		if len(ipv4PodNetworks) > 1 {
+			return fmt.Errorf("exactly one v4 pod network is supported. v4 pod networks: %s", ipv4PodNetworks)
+		}
+		ipv4ServiceNetworks := network.GetByIPFamily(cfg.ServiceNetworks, network.IPv4Family)
+		if len(ipv4ServiceNetworks) > 1 {
+			return fmt.Errorf("exactly one v4 service network is supported. v4 service networks: %s", ipv4ServiceNetworks)
+		}
+		ipv4NodeNetworks := network.GetByIPFamily(cfg.NodeNetworks, network.IPv4Family)
+		if len(ipv4NodeNetworks) > 1 {
+			return fmt.Errorf("exactly one v4 node network is supported. v4 node networks: %s", ipv4NodeNetworks)
+		}
+
+		for _, nw := range ipv4PodNetworks {
+			err = ipTable.AppendUnique("nat", "OUTPUT", "-m", "owner", "--uid-owner", constants.EnvoyNonRootUserId, "-d", nw.String(), "-j", "NETMAP", "--to", constants.ShootPodNetworkMapped)
+			if err != nil {
+				return err
+			}
+		}
+		for _, nw := range ipv4ServiceNetworks {
+			err = ipTable.AppendUnique("nat", "OUTPUT", "-m", "owner", "--uid-owner", constants.EnvoyNonRootUserId, "-d", nw.String(), "-j", "NETMAP", "--to", constants.ShootServiceNetworkMapped)
+			if err != nil {
+				return err
+			}
+		}
+		for _, nw := range ipv4NodeNetworks {
+			err = ipTable.AppendUnique("nat", "OUTPUT", "-m", "owner", "--uid-owner", constants.EnvoyNonRootUserId, "-d", nw.String(), "-j", "NETMAP", "--to", constants.ShootNodeNetworkMapped)
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
Do we want to unify this into a single loop? Please also add error context.

Reply:
we keep it this way for easier readability with a little more code.